src/OpenTelemetry.Instrumentation.Wcf/Implementation/TracingErrorHandler.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NETFRAMEWORK
+using System.Diagnostics;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Dispatcher;
+
+namespace OpenTelemetry.Instrumentation.Wcf.Implementation;
+
+internal class TracingErrorHandler : IErrorHandler
+{
+    public bool HandleError(Exception error)
+    {
+        return false;
+    }
+
+    public void ProvideFault(Exception error, MessageVersion version, ref Message fault)
+    {
+        // By rights this should be in `HandleError` instead, which would keep it from
+        // interfering with the response to the client.
+        // However, by the time `HandleError` fires, the Activity has already been stopped
+        // so the error appears after the Activity has completed.  Additionally, sometimes
+        // the context has already been disposed or otherwise lost, preventing association
+        // at all.
+        // Also it becomes very difficult to unit-test, because there is no easy `ErrorsHandled`
```

Comment:
This is perfectly fine to do here. HandleError allows you to prevent the session/context/connection from being aborted, and ProvideFault enable you to provde a custom fault message. Neither one of them are designed specifically for what you're doing, so there's no "right" place to do this. HanndleError is no more the right place than ProvideFault, so if this method works better due to the activity being stopped, then it's the right method to use.